### DISCUSSION TITLE
When run 'patrol test', the app installs but doesn't start

### DISCUSSION THREAD

**Discussion by jmvalenciz** [leancodepl/patrol#1183] [Q&A]
Hi there.

I'm configuring Patrol with Native automation (I already finished the setup guide) in one of my flutter projects and I'm facing an error (I configured a [minimal example](https://github.com/jmvalenciz/patrol_test) of the project):

When I execute `patrol test`, the app installs correctly in the device but it doesn't start by itself and `patrol_cli` gets stuck indefinitely so I decided to look at the logs and this is what I found:

# Logs
Command: `flutter logs`
Output:
```
Showing 2201117SG logs:
V/FlutterTestRunner(29567): launchActivity failed, possibly because the activity was already running. java.lang.RuntimeException: Could not launch intent Intent { act=android.intent.action.MAIN flg=0x14000000 cmp=com.example.test_patrol/.MainActivity } within 45000 milliseconds. Perhaps the main thread has not gone idle within a reasonable amount of time? There could be an animation or something constantly repainting the screen. Or the activity is doing network calls on creation? See the threaddump logs. For your reference the last time the event queue was idle before your activity launch request was 1680729249728 and now the last time the queue went idle was: 1680729249728. If these numbers are the same your activity might be hogging the event queue.
V/FlutterTestRunner(29567): Try disabling auto-launch of the activity, e.g. ActivityTestRule<>(MainActivity.class, true, false);
```

Command: `adb logcat | grep -i PatrolServer`
Output:
```
04-05 16:14:09.776 29567 29586 I PatrolServer: Starting server loop...
04-05 16:14:09.868 29567 29586 I PatrolServer: Starting server...
04-05 16:14:09.871 29567 29586 I PatrolServer: Server started on http://localhost:8081
04-05 16:14:54.902 29567 29586 E THREAD_STATE:     pl.leancode.patrol.PatrolServer.blockUntilShutdown(PatrolServer.kt:39)
```

# Execution environment
patrol version: ^1.0.8
patrol_cli version: v1.1.3
`flutter --version`: 
```
Flutter 3.7.8 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 90c64ed42b (2 weeks ago) • 2023-03-21 11:27:08 -0500
Engine • revision 9aa7816315
Tools • Dart 2.19.5 • DevTools 2.20.1
```
OS: Arch Linux
Android Device: Xiaomi Redmi Note 11S (2201117SG)

# Extra data
I also tried to start the app manually from the `adb shell` with the command `am start -n com.example.test_patrol/.MainActivity` or just clicking the app icon and it works as expected.

**Comment by AlviaNabil** [comment]
@jmvalenciz Facing the same issue on 2201117SI device. Have you found any solution?

**Comment by qube-qa-sonu** [comment]
facing the same issue. Is anybody found the solution?
![image](https://github.com/leancodepl/patrol/assets/146061190/acc60c55-ed83-4904-ab22-f4ac96a4e70f)
application builded, installed in phone. test passed in terminal. but the app did not launched in phone.

`import 'dart:io';

import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';
import 'package:patrol/patrol.dart';
import 'package:integration_test/integration_test.dart';
import 'package:myApp/main.dart';

void main() {
  patrolTest(
    'counter state is the same after going to home and switching apps',
    ($) async {
      await $.pumpWidgetAndSettle(const MyApp());
      expect($, findsNothing);
    },
  );
}`

`patrol test -t integration_test\example_test.dart`

**Reply by qube-qa-sonu** [reply]
1. Please verify that the location of mentioned folder is correct: android\app\src\main\kotlin\com\your_package\.

2. Please ensure that the following files contain the correct package name:
- android\app\src\debug\AndroidManifest.xml
- android\app\src\main\AndroidManifest.xml
- android\app\src\profile\AndroidManifest.xml
- android\app\src\main\kotlin\com\your_package\FileUtil.kt
- android\app\src\main\kotlin\com\your_package\MainActivity.kt
- android\app\src\main\kotlin\com\your_package\MainApplication.kt
- android\app\src\main\kotlin\com\your_package\RootUtil.kt
- android\app\src\main\kotlin\com\your_package\ScannerActivity.kt
- android\app\src\main\kotlin\com\your_package\ScannerOverlay.kt

3. Please verify that the example_test.dart file contains the necessary configurations required to launch the app, similar to those found in main.dart. You can seek assistance from a developer if needed.
For example:
- Just like dev included the runApp() function with the required parameters in main.dart.
- Include the $pumpWidgetAndSettle() function with the required parameters in example_test.dart.

I've completed these tasks. I realize it took some time— you can check the date of my last comment— but now my code is functioning properly with Patrol.

**Comment by adrian-moisa** [comment]
I'm facing the same problem on latest patrol 3.6.1, SDK 34 and gradle 8.7. Standalone app starts fine on both native and web.

**Reply by AlviaNabil** [reply]
Can you tell me which device are you using? And what operating system you are running?

**Reply by adrian-moisa** [reply]
Mac M1. Found the issue by reading advice from #1629 and #1631 . I noticed in git that `MainActivityTest.java` was labeled as renamed. I found this label to be intriguing because the name looked the same. After a very careful inspection I found that there was a whitespace in the name after the extension. I'm not even sure why this is permitted. Somehow it snuck there. After fixing it the tests started working again. I suggested that maybe we can have a check to warn users if the `MainActivityTest.java` is not found as expected. That would save some time.

**Comment by ashwinkey04** [comment]
I'm on Apple silicon as well, trying to run patrol tests for an android only flutter app. I redid the whole setup twice to ensure I don't do typos or skip steps by mistake. Yet poth `patrol test -t integration_test/example_test.dart --device=emulator-5554 --flavor dev` and its corresponding `patrol develop` command always builds the test app successfully but never launches it on the emulator. 

Here are some relevant code snips: 
- [Java and Kotlin files](https://hastebin.com/share/anobegexen.kotlin)
- [example_test.dart](https://hastebin.com/share/edekidayij.dart)
- [Folder structure screenshot](https://cdn.discordapp.com/attachments/1241783517877108736/1242395650520256673/Screenshot_2024-05-21_at_2.06.02_PM.png?ex=664dae9d&is=664c5d1d&hm=3cf6b78fd5258e3efeb3a86aa9406d6de69e41ea80af7807610e8e2c3ed00614&)

These are the last few lines of patrol test with verbose
![image](https://github.com/leancodepl/patrol/assets/20596763/78a7bd26-fba7-4fac-9063-bcda414a2d33)
